x-pack/plugins/endpoint/public/applications/endpoint/view/use_page_id.ts

```diff
@@ -0,0 +1,20 @@
+/*
```

Comment:
In the past, we relied on `react-router` to tell us when a user entered a page and we would then fire an action to let the app know about this navigation. The approach below is simpler. We use a hook that is meant to be called from our app's main pages or containers. When the user navigates to these pages, the hook will run and will dispatch a `userNavigatedToPage` action. Our middleware can then listen for this action and perform side-effects like API calls. Thoughts @oatkiller @kevinlog @paul-tavares @parkiino?

Reply:
ya let's keep it simple. We could end up only having 2 pages (or less) for a long time.

Reply:
I'm ok with this approach as well.
We should also have a store action that indicates a user has left the page (`userNavigatedAwayFrompage`) so that proper clean up (ex. reset store namespace to clean out data loaded in memory).